### DISCUSSION TITLE
make query on array of object

### DISCUSSION THREAD

**Discussion by AliAmini78** [jenssegers/laravel-mongodb#2479] [Q&A] [answered]
- Laravel Version: 9
- PHP Version: 8.1
- jenssegers/mongodbn: 3.9

At my model there's a field that represents an array of object, actually are instoke of my record, for example:

```
 { "item": "journal", "instock": [ { "warehouse": "A", "qty": 5 }, { "warehouse": "C", "qty": 15 } ] },
    { "item": "notebook", "instock": [ { "warehouse": "C", "qty": 5 } ] },
    { "item": "paper", "instock": [ { "warehouse": "A", "qty": 60 }, { "warehouse": "B", "qty": 15 } ] },
    { "item": "planner", "instock": [ { "warehouse": "A", "qty": 40 }, { "warehouse": "B", "qty": 5 } ] },
    { "item": "postcard", "instock": [ { "warehouse": "B", "qty": 15 }, { "warehouse": "C", "qty": 35 } ] }

```
I would retrieve all record that " warehouse=A and qty > 50  " .
It's possible?

**Comment by Giacomo92** [comment, marked as answer]
Hi @AliAmini78 ,
Have you tried this query?
`Model::where('instock.warehouse', 'A')->where('instock.qty', '>', 50)->get()`

**Reply by AliAmini78** [reply]
This worked !!
thanks for your help✌️🙏
